### PR TITLE
feat: Page tagging

### PR DESCRIPTION
**Pull request by limhjgrace**
Users want to add tags to pages so that data from RUM events can be separated into different groups for analysis, metric aggregation, etc. Currently, users are unable to add tags to pages. This feature will allow them to add tags and will store the page tag data in the event metadata. Users can pass in the page tag data:
(1) when manually recording page view events using `recordPageView()`
(2) by manually setting the page tag data using `setCustomAttributes()`

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by qhanam** [comment]
Could we separate the page tagging change from the custom attributes change?

Specifically, I'm not sure we want to add `setCustomAttributes` with this change. While it is clear how the application will add custom page attributes (i.e., using `recordPageView` + `PageAttributes`), we'll need to think about how the application will add custom session attributes (i.e., using the `config` object or using some `addSessionAttribute` function).

File: src/CommandQueue.ts
Comment (limhjgrace) [reply src/CommandQueue.ts:41]: Refactored

File: src/sessions/PageManager.ts
Comment (limhjgrace) [reply src/sessions/PageManager.ts:146]: Removed and refactored

Comment (limhjgrace) [reply src/sessions/PageManager.ts:146]: Removed and refactored

**Comment by qhanam** [comment]
Looks good to me! Let's wait until the event schema is deployed to merge.

**Review by qhanam — APPROVED** [review]

**Pull request merged**
